includes/WikiDiscover.php

```diff
@@ -117,111 +108,159 @@ public function isInactive( $database ) {
 	public function isPrivate( $database ) {
 		return in_array( $database, $this->private );
 	}
+	
+	/**
+	 * @param Parser $parser
+	 */
+	public static function onParserFirstCallInit( Parser $parser ) {
+		$parser->setFunctionHook( 'numberofwikisincategory', [ __CLASS__, 'numberOfWikisInCategory' ], Parser::SFH_NO_HASH );
+		$parser->setFunctionHook( 'numberofwikisinlanguage', [ __CLASS__, 'numberOfWikisInLanguage' ], Parser::SFH_NO_HASH );
+		$parser->setFunctionHook( 'numberofwikisbysetting', [ __CLASS__, 'numberOfWikisBySetting' ], Parser::SFH_NO_HASH );
+	}
+	
+	/**
+	 * @param Parser $parser
+	 * @param String $category|null
+	 * @return integer
+	 */
+	public static function numberOfWikisInCategory( Parser $parser, String $category = null ) {
+		$config = MediaWikiServices::getInstance()->getMainConfig();
+
+		$dbw = wfGetDB( DB_MASTER, [], $config->get( 'CreateWikiDatabase' ) );
+
+		return $dbw->selectRowCount( 'cw_wikis', '*', [ 'wiki_deleted' => 0, 'wiki_category' => strtolower( $category ) ] );;
+	}
+
+	/**
+	 * @param Parser $parser
+	 * @param String $language|null
+	 * @return integer
+	 */
+	public static function numberOfWikisInLanguage( Parser $parser, String $language = null ) {
+		$config = MediaWikiServices::getInstance()->getMainConfig();
+
+		$dbw = wfGetDB( DB_MASTER, [], $config->get( 'CreateWikiDatabase' ) );
 
+		return $dbw->selectRowCount( 'cw_wikis', '*', [ 'wiki_deleted' => 0, 'wiki_language' => strtolower( $language ) ] );;
+	}
+	
+	/**
+	 * @param Parser $parser
+	 * @param mixed $setting|null
+	 * @param mixed $value|null
+	 * @return string|integer
+	 */
+	public static function numberOfWikisBySetting( $parser, $setting = null, $value = null ) {
+		if ( !$setting && !$value ) {
+			return 'Error: no input specified.';
+		}
+
+		$config = MediaWikiServices::getInstance()->getMainConfig();
+		$dbw = wfGetDB( DB_MASTER, [], $config->get( 'CreateWikiDatabase' ) );
+
+		$mwExt = new ManageWikiExtensions( $config->get( 'DBname') );
+		$extList = $mwExt->list();
+        
+		if ( !$value && !in_array( $setting, $extList ) ) {
+			return 0;
+		}
+        
+		if ( in_array( $setting, $extList ) ) {
+			$s_extensions = (array)$dbw->selectField( 'mw_settings', 's_extensions' );
+			$res = $dbw->select( 'mw_settings', 's_extensions' );
+			$extensionUsageCount = 0;
+
+			foreach ( $res as $row ) {
+				if( in_array( $setting, array_flip( $s_extensions ) ) ) {
```

Comment:
You're foreaching using $res, but you're not using it inside the foreach which is a waste of resources

Reply:
fixed, no longer foreaching at all here.